### PR TITLE
[AGENT-4318] Implement UploadDatasetOperator

### PR DESCRIPTION
**Pull request by oleksandr-saienko**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
Added new Airflow Operator for Uploading local file to DataRobot AI Catalog
Bump version to 0.0.5
## Rationale
Uploading local file to DataRobot AI Catalog is important missing part in out Airflow Provider

### CODE REVIEW

File: datarobot_provider/operators/ai_catalog.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    template_fields: Iterable[str] = []
+    template_fields_renderers: Dict[str, str] = {}
+    template_ext: Iterable[str] = ()
+    ui_color = '#f4a460'
+
+    def __init__(
+            self,
+            *,
+            datarobot_conn_id: str = "datarobot_default",
+            **kwargs: Any,
```
Comment (andrius-senulis) [inline datarobot_provider/operators/ai_catalog.py:44]: overindented

**Review by andrius-senulis — CHANGES_REQUESTED** [review]
Since we don't have automation in the repo yet, can you please manually run lint, black, isort, typecheck, check-licenses ([defined in Makefile](https://github.com/datarobot/airflow-provider-datarobot/blob/main/Makefile))?

File: README.md [hunk 2]
```diff
@@ -214,6 +214,16 @@ in the `context["params"]` variable, e.g. getting a training data you would use
 
     Returns a dict with the feature drift data.
 
+- `UploadDatasetOperator`
```
Comment (andrius-senulis) [inline README.md:217]: I would place it as the first operator in the list, as it would fit the logical flow of actions in DR pipeline.

**Review by andrius-senulis — APPROVED** [review]
One minor comment, otherwise LGTM.

**Review by elatt — CHANGES_REQUESTED** [review]
Let's get the Jenkins infra (#11) merged in first.

**Comment by elatt** [comment]
Jenkins run

**Review by elatt — APPROVED** [review]

**Pull request merged**